storage/netapp_e/na_eseries_storage_system.py

```diff
@@ -0,0 +1,319 @@
+    elif rc == 404:
+        if state == 'present':
+            changed = True
+            array_exists = False
+        else:
+            changed = False
+            module.exit_json(changed=changed, msg="Storage system was not present.")
+
+    if changed and not module.check_mode:
+        if state == 'present':
+            if not array_exists:
+                # add the array
+                array_add_req = dict(
+                    id=ssid,
+                    controllerAddresses=controller_addresses,
+                    metaTags=meta_tags,
+                    enableTrace=enable_trace
+                )
+
+                if array_wwn:
+                    array_add_req['wwn'] = array_wwn
+
+                if array_password:
+                    array_add_req['password'] = array_password
+
+                post_headers = dict(Accept="application/json")
+                post_headers['Content-Type'] = 'application/json'
+
+                try:
+                    request_data = json.dumps(array_add_req)
+                    (rc, resp) = request(api_url + "/storage-systems", data=request_data, headers=post_headers,
+                                         method='POST', url_username=api_usr, url_password=api_pwd,
+                                         validate_certs=validate_certs)
+                except:
+                    err = get_exception()
+                    module.fail_json(
+                        msg="Error adding storage array to proxy. Request body [{0}]. Error[{1}].".format(request_data,
+                                                                                                      str(err)))
+
+                if rc == 201:
+                    status = 'neverContacted'
+                    fail_after_time = dt.utcnow() + timedelta(seconds=array_status_timeout_sec)
+
+                    while status == 'neverContacted':  # TODO: watch for other states?
+                        if dt.utcnow() > fail_after_time:
+                            raise Exception("web proxy timed out waiting for array status")
+
+                        sleep(1)
+                        (rc, system_resp) = request(api_url + "/storage-systems/{0}".format(ssid),
+                                                    headers=dict(Accept="application/json"), url_username=api_usr,
+                                                    url_password=api_pwd, validate_certs=validate_certs,
+                                                    ignore_errors=True)
+                        status = system_resp['status']
+
+            else:  # array exists, modify...
+                post_headers = dict(Accept="application/json")
+                post_headers['Content-Type'] = 'application/json'
+                post_body = dict(
+                    controllerAddresses=controller_addresses,
+                    removeAllTags=True,
+                    enableTrace=enable_trace,
+                    metaTags=meta_tags
+                )
+
```

Comment:
I refactored the common lines into a function.  Across all the other modules I'll be sending across there is *a lot* of refactoring that can be done to remove technical debt like this.  I have it on our ansible backlog because I'm under pressure to get all the modules available to the community.